ale_linters/perl/perl.vim

```diff
@@ -27,12 +27,20 @@ function! ale_linters#perl#perl#Handle(buffer, lines) abort
     let l:output = []
     let l:basename = expand('#' . a:buffer . ':t')
 
+    let type = 'E'
+    let syntax_ok = matchstr(a:lines[-1], 'syntax OK')
+    if syntax_ok != ''
+        let type = 'W'
+    endif
+
+    let seen = {}
+
     for l:match in ale#util#GetMatches(a:lines, l:pattern)
         let l:line = l:match[3]
         let l:text = l:match[1]
-        let l:type = 'E'
+        let l:type = type
 
-        if ale#path#IsBufferPath(a:buffer, l:match[2])
+        if ( ale#path#IsBufferPath(a:buffer, l:match[2]) && !has_key(seen,l:line) )
```

Comment:
You don't need the parentheses here. There's already another condition on the line below, so you could maybe move your second check to a new line below.

Reply:
Fixed in 43782df621315ad